tests/framework/utils.py

```diff
@@ -760,17 +761,29 @@ def start_screen_process(screen_log, session_name, binary_path, binary_params):
         delay=1,
     ).group(1)
 
-    binary_clone_pid = int(
-        open("/proc/{0}/task/{0}/children".format(screen_pid), encoding="utf-8")
-        .read()
-        .strip()
-    )
+    # Make sure the screen process launched successfully
+    # As the parent process for the binary.
+    screen_ps = psutil.Process(int(screen_pid))
+    for _ in range(5):
+        if not screen_ps.is_running():
+            time.sleep(1)
+        else:
+            break
```

Comment:
You could add an `else` here to print something 

```python
for ...
else:
    print("screen_ps still not running. bailing out after 5 tries and hoping for the best")

Reply:
The approach of using the `@retry wait_process_running()` will log a warning for each attempt that fails.

Reply:
Indeed.